openerp/http.py

```diff
@@ -1361,12 +1363,18 @@ def session_gc(session_store):
     if random.random() < 0.001:
         # we keep session one week
         last_week = time.time() - 60*60*24*7
-        for fname in os.listdir(session_store.path):
-            path = os.path.join(session_store.path, fname)
+        if isinstance(session_store, werkzeug.contrib.sessions.FilesystemSessionStore):
+            for fname in os.listdir(session_store.path):
+                path = os.path.join(session_store.path, fname)
+                try:
+                    if os.path.getmtime(path) < last_week:
+                        os.unlink(path)
+                except OSError:
+                    pass
+        else:
             try:
-                if os.path.getmtime(path) < last_week:
-                    os.unlink(path)
-            except OSError:
+                session_store.session_gc(last_week)
+            except AttributeError:
```

Comment:
This is not needed.
Separate garbage collection of uploaded wizard files and sessions was correct.
The case being, when the sessions in Redis (for example) expired, on their own, we will have no recollection of what uploaded files needed to be garbage collected, there is no notification callback.

Better would be when the session expiration needed to be extended, touch() their corresponding uploaded files.
The method `touch()` should be in that custom SessionStore base class.